messgen/ts_generator.py

```diff
@@ -12,150 +16,166 @@
+
+        if self.imports:
+            code += self.generate_imports(out_dir, output_path)
+
+        code += '\n'.join(self.code_lines)
+
+        with open(output_file, 'w', encoding='utf-8') as f:
+            f.write(code)
+
+
+    def generate_imports(self, out_dir, output_path):
+        imports = ''
+
+        for proto_name, type_name in sorted(self.imports):
+            import_path = os.path.relpath(
+                os.path.join(out_dir, proto_name.replace(SEPARATOR, os.sep), proto_name.split(SEPARATOR)[-1]),
+                output_path
+            ).replace('\\', '/')
+
+            imports += f'import {{ {self.to_camel_case(type_name)} }} from "{import_path}";\n'
+        return imports + '\n'
+    
+    
+    def generate_type(self, proto_name, type_name, types):
+        if (proto_name, type_name) in self.generated_types:
+            return
+        self.generated_types[(proto_name, type_name)] = True
+
+        type_def = types[type_name]
+        type_class = type_def.get("type_class")
+        comment = type_def.get("comment", "")
+        interface_name = self.to_camel_case(type_name)
+
+        if type_class == "struct":
+            fields = type_def.get("fields") or []
+            for field in fields:
+                field_type = field["type"]
+                base_type = self.get_base_type(field_type)
+                if not self.is_builtin_type(base_type):
+                    self.handle_custom_type(base_type, proto_name)
+        elif type_class == "enum":
+            pass
+        else:
+            pass  
+
+        if comment:
+            self.code_lines.append(f"/** {comment} */")
+
+        if type_class == "struct":
+            self.code_lines.append(f"export interface {interface_name} {{")
+            fields = type_def.get("fields") or []
+            for field in fields:
+                field_comment = field.get("comment", "")
+                field_name = field["name"]
+                field_type = field["type"]
+                ts_type = self.get_ts_type(field_type, proto_name)
+                if field_comment:
+                    self.code_lines.append(f"  /** {field_comment} */")
+                self.code_lines.append(f"  {field_name}: {ts_type};")
+            self.code_lines.append("}\n")
+        elif type_class == "enum":
+            self.code_lines.append(f"export enum {interface_name} {{")
+            values = type_def.get("values", [])
+            for value in values:
+                name = value["name"]
+                value_value = value["value"]
+                value_comment = value.get("comment", "")
+                if value_comment:
+                    self.code_lines.append(f"  /** {value_comment} */")
+                self.code_lines.append(f"  {name} = {value_value},")
+            self.code_lines.append("}\n")
+        else:
+            pass  
+
+    def get_ts_type(self, field_type, current_proto_name):
+        # Обработка массивов, карт и вложенных типов
```

Comment:
I think it would be beneficial to use English for comments as it would be understandable for a wider audience.

Reply:
yeah sure, sorry, i thought i opened a pull request in draft, there are a couple of things i wanted to fix, but thanks a lot for the review